### ISSUE TITLE
Authentication issue when trying to use search_fullarchive

### ISSUE THREAD

**Issue by kmcmorrow1** [ropensci/rtweet#569]
### Problem

Despite having a premium account (and properly setting up the dev environment for the Twitter API -- see screenshot below) I still run into authentication issues when using the function "search_fullarchive()"

Things I have tried

- Looking through existing issues on here
- Renewing my consumer/access keys and secrets
- Deleting and recreating the app / dev enviroment

### Reproduce the problem

``` r
## insert code here
token = rtweet::create_token(app = "BR_App123", consumer_key = "5w0opka1N******pMlI7iIRWxtUt",
                             consumer_secret = "DGjrqmS0fWC7OQbh5EMFp9g******zQ85cIvjdegd867rLQ",
                             access_token = "8290564******8-YxDH2UlklXtRU90N0Jwh7PrRBayHaxu",
                             access_secret = "tGBo3jJrG******FR4671zlocnWZf9aLboEsyio6FQR6")

search_fullarchive(q = "test", env_name = "development", n=10, fromDate = "201912120000", toDate = "201912120000", token=token)

```

### rtweet version

``` r
## copy/paste output
packageVersion("rtweet")
```
‘0.7.0’

### Session info

``` r
## copy/paste output
sessionInfo()
```
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] openssl_1.4.3 rtweet_0.7.0  httpuv_1.5.4  httr_1.4.2   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5         later_1.1.0.1      cellranger_1.1.0   compiler_4.0.2     pillar_1.4.7      
 [6] RColorBrewer_1.1-2 prettyunits_1.1.1  progress_1.2.2     bitops_1.0-6       tools_4.0.2       
[11] digest_0.6.27      jsonlite_1.7.1     lifecycle_0.2.0    tibble_3.0.6       pkgconfig_2.0.3   
[16] rlang_0.4.10       cli_2.3.0          curl_4.3           binman_0.1.1       xfun_0.18         
[21] dplyr_1.0.2        stringr_1.4.0      hms_1.0.0          htmlwidgets_1.5.2  generics_0.0.2    
[26] vctrs_0.3.6        askpass_1.1        caTools_1.18.0     DT_0.15            wordcloud_2.6     
[31] tidyselect_1.1.0   glue_1.4.2         R6_2.5.0           XML_3.99-0.5       readxl_1.3.1      
[36] RSelenium_1.7.7    purrr_0.3.4        semver_0.2.0       magrittr_2.0.1     promises_1.1.1    
[41] ellipsis_0.3.1     htmltools_0.5.0    assertthat_0.2.1   tinytex_0.26       stringi_1.5.3     
[46] wdman_0.2.5        crayon_1.4.1   

<img width="452" alt="Screen Shot 2021-04-21 at 10 16 28 AM" src="https://user-images.githubusercontent.com/33642859/115594489-a73e5a00-a28a-11eb-9a43-39e338b1d022.png">

**Comment by llrs** [comment]
Hi,
Unfortunately without any error message or further indications of what "authentication issues when using the function "search_fullarchive()" mean I cannot provide help. You don't mention if you have run into connectivity problems or with problems with other endpoints when the `search_fullarchive()` function failed you, does this mean it just stops working without error message or there is some error message of failing to authenticate in the middle of a function?

Without more information I can only point to use the development version of the package, where authentication has been reviewed and updated.

**Comment by kmcmorrow1** [comment]
> Hi,
> Unfortunately without any error message or further indications of what "authentication issues when using the function "search_fullarchive()" mean I cannot provide help. You don't mention if you have run into connectivity problems or with problems with other endpoints when the `search_fullarchive()` function failed you, does this mean it just stops working without error message or there is some error message of failing to authenticate in the middle of a function?
> 
> Without more information I can only point to use the development version of the package, where authentication has been reviewed and updated.

Sorry about that, completely forgot to attach the error message. Here is the error that gets thrown when I try to use the function: 

"tweets/search/fullarchive/development ***Warning: list(message = "Forbidden: Authentication succeeded but account is not authorized to access this resource.", sent = "2021-04-21T21:55:05+00:00", transactionId = "00006a8d0013c064")
data frame with 0 columns and 0 rows".

In regards to any other issues I've experienced, I am able to use the API/other rtweet functions (like 'search_tweets') with the same token, and I experience no issues.

**Comment by llrs** [comment]
Use `rate_limit()` and search for tweets/search/fullarchive/development. Do you have some remaining API calls?

**Comment by kmcmorrow1** [comment]
> Use `rate_limit()` and search for the correct endpoint. Do you have some remaining API calls?

Would you be able to expand on what you mean by "correct endpoint"? I'm not quite sure what I should be looking for here. The only query that has a "remaining" value that is less than the corresponding "limit" is: "application/rate_limit_status", and even then there are 178 remaining.

As for API calls, I have called both search_tweets and search_fullarchive maybe a cumulative 10 times, so I should have plenty of API calls left

**Comment by llrs** [comment]
The endpoint is where the requests are made on twitter. You should look for "tweets/search/". 
It is a bit weird that you made some calls with search_tweets and you don't see the remaining value less than the limit. Perhaps you made them a while back? Please check run the `search_fullarchive()` and the `rate_limit()` in less than 15 minutes. 

But if you say there are the same of remaining than on limit then it might be two other things: You are using a token that has not premium access or you made more requests than allowed per time:
Paid subscription requests / minute: 60
Sandbox subscription requests / minute: 30
Both paid and Sandbox requests / second: 10

I'm closing the issue as it doesn't seem to be a bug on rtweet.

**Comment by kmcmorrow1** [comment]
> But if you say there are the same of remaining than on limit then it might be two other things: You are using a token that has not premium access or you made more requests than allowed per time:

I'm confused. Aren't you able to use the sandbox version (non-premium) to utilize the 30-days and full archive search functionalities? I haven't paid for the premium access of the API, but according to the Twitter dev environment page, I should still be able to utilize both of those functions.

As for the request limit, I have double checked, and I am not exceeding the sandbox limits for requests (in part because the requests are not working with rtweet.)

**Comment by llrs** [comment]
I haven't set the new testing user with the sandboxes (yet); so yes, I'm unable to test the 30-days and full archive search functionalities myself at the moment. 

In the last 2 years the package has been on version 0.7.0 there aren't other reports of something similar, so I think this might be a problem specific to you or your token. I reopen the issue to remind me when I review the premium functionality to look it up, but it might take a while.

**Comment by kmcmorrow1** [comment]
I'm not sure what changed but this morning I restarted my machine and re-opened R, and **now the search_fullarchive function works as expected.** I wish I could offer an explanation as to why it suddenly works (because I didn't change anything!) but either way, thank you for the help.

**Comment by llrs** [comment]
Glad it is finally working for you!

**Comment by llrs** [comment]
Closing this now as this was fixed and the new PR #716 should work and now I can test it (even if only locally with the premium API)
